ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/HikariJdbcConnectionProvider.scala

```diff
@@ -13,48 +13,40 @@ import com.daml.ledger.resources.{Resource, ResourceContext, ResourceOwner}
 import com.daml.logging.{ContextualizedLogger, LoggingContext}
 import com.daml.metrics.{DatabaseMetrics, Timed}
 import com.daml.platform.configuration.ServerRole
-import com.daml.platform.store.DbType
-import com.daml.platform.store.appendonlydao.HikariJdbcConnectionProvider._
 import com.daml.timer.RetryStrategy
 import com.zaxxer.hikari.{HikariConfig, HikariDataSource}
+import javax.sql.DataSource
 
 import scala.concurrent.Future
 import scala.concurrent.duration.{DurationInt, FiniteDuration}
 import scala.util.control.NonFatal
 
-private[platform] final class HikariConnection(
+private[platform] final class HikariDataSourceOwner(
+    dataSource: DataSource,
     serverRole: ServerRole,
     jdbcUrl: String,
     minimumIdle: Int,
     maxPoolSize: Int,
     connectionTimeout: FiniteDuration,
     metrics: Option[MetricRegistry],
-    connectionPoolPrefix: String,
-    maxInitialConnectRetryAttempts: Int,
-    connectionAsyncCommitMode: DbType.AsyncCommitMode,
+    connectionPoolPrefix: String = "daml.index.db.connection",
+    maxInitialConnectRetryAttempts: Int = 600,
 )(implicit loggingContext: LoggingContext)
-    extends ResourceOwner[HikariDataSource] {
+    extends ResourceOwner[DataSource] {
 
   private val logger = ContextualizedLogger.get(this.getClass)
 
   override def acquire()(implicit context: ResourceContext): Resource[HikariDataSource] = {
     val config = new HikariConfig
-    val dbType = DbType.jdbcType(jdbcUrl)
-
+    config.setDataSource(dataSource)
     config.setJdbcUrl(jdbcUrl)
```

Comment:
Do you need to set both the dataSource and the jdbcUrl?

Reply:
of course not! so nice, more dependencies removed :tada: